### PR TITLE
Put in Extractor with DBPediaSpotlight code.

### PR DESCRIPTION
**Pull request by XuChen-Repair**
Delete DBPedia Spotlight written in Java. Add in Python version

### CODE REVIEW

**Comment by shubhamgoyal** [comment]
@XuChen-Repair write description about what this pull request is about exactly. For an example, see this  - https://github.com/SeSaMe-NUS/Trendspedia2/pull/12

**Comment by shubhamgoyal** [comment]
@allwinpb please review

**Comment by allwinpb** [comment]
@XuChen-Repair I really need a description to see what you're doing haha. Plus it is better to add `*.class` files into gitignore and not commit those files to the repo.

File: DBPediaSpotlight/src/Extractor.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	
+	public List<String> getNameList() {
+		db.configiration(0.0, 0, "non", "CoOccurrenceBasedSelector", "Default", "yes");
+	    try {
+			db.evaluate(query);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	    
+	    List<String> temp = db.getResu();
```
Comment (allwinpb) [inline DBPediaSpotlight/src/Extractor.java]: Is this a typo? And it still ran without problems?

Comment (XuChen-Repair) [inline DBPediaSpotlight/src/Extractor.java]: I don't think there is a typo. But anyway, I am going to change the code
into python. Will let you know soon. I will avoid the mistakes I have made
next time when I submit a pull request. Thank you.

2015-07-16 13:50 GMT+08:00 Allwin <notifications@github.com>:

**Comment by XuChen-Repair** [comment]
Delete DBPedia Spotlight written in Java. Add in Python version

**Comment by shubhamgoyal** [comment]
@allwinpb code review please :)

**Comment by allwinpb** [comment]
@XuChen-Repair 

I'm guessing you are new at python. You might want to get more acquainted with it before contributing. There is something called `pip` for python. With this tool (It's bundled with python in most cases, if it isn't it's a simple enough google search), you can "link" to other python libraries.

For example if you want to use the "pyspotlight" library as in this case, all you need to do is to go to a terminal and type `pip install pyspotlight`. Awesome right? No need to commit the entire library. This way, most of the code in our repo will be code that we authored ourselves. It also reduces the size of the repository. In this case we can get rid of the entire `pyspotlight-master` folder.

Also a "wrapper" library is something that wraps around an existing library or service. In this case, all `pyspotlight` does is to provide a python interface (so python programs can access functions of the service) to the original Java library (DBpedia-Spotlight). So we need the original library as well. The `pyspotlight` README file has more information on how to use the wrapper library.

Take your time and read up on `pip`. If possible also read up on `virtualenv`. They will save you tons of time in the future. Once you feel comfortable with python and pip in general, then we can get started.

**Comment by XuChen-Repair** [comment]
Yes, I am a beginner. I will get more acquainted with it before contributing.

Best Regards,
Xu Chen

**Comment by shubhamgoyal** [comment]
@XuChen-Repair @allwinpb is right and you would do well to note his suggestions. Very useful advice for software engineering in Python :)
Anyways, feel free to open pull requests. If something is wrong, we will point it out and make you re-open pull request anyways haha

**Pull request closed without merging**